corehq/elastic.py

```diff
@@ -212,7 +212,10 @@ def run_query(index_name, q, debug_host=None, es_instance_alias=ES_DEFAULT_INSTA
         else:
             raise
     try:
-        return es_instance.search(es_meta.index, es_meta.type, body=q)
+        results = es_instance.search(es_meta.index, es_meta.type, body=q)
+        if results.get('_shards', {}).get('failed'):
```

Comment:
curious if you were able to test this against a local shard failure. This is the type of code I would get something wrong with if I were just doing it based on my reading of what the docs say should be returned

Reply:
This would be hard to test against a local shard failure as I'd have to somehow bring down a single shard. Locally, this would mean spinning up a second elasticsearch node and adding it to my local cluster, then causing that node to fail.

I'm fairly confident that all ES search results have the `_shards` object returned with a `failed` key/value pair. 

I'd be happy to try and simulate this if you think it is important, though.

Reply:
I think as long as you're sure the `search` method never returns `None` or a non-`dict`, I don't see how this could cause problems. (The worst I can imagine is that a mistake here means we don't log these correctly.)